### DISCUSSION TITLE
appearance-none utility does not hide input number arrows

### DISCUSSION THREAD

**Discussion by bswinnerton** [tailwindlabs/tailwindcss#2173] [Ideas]
### Describe the problem:

When using an `input` of type `number` with the `appearance-none` utility, browser-based arrows still appear. This seems to be counter to the description of [`.appearance-none`](https://tailwindcss.com/docs/appearance#class-reference), which is described as:

> Use `.appearance-none` to reset any browser specific styling on an element. 

### Link to a minimal reproduction:

https://codepen.io/bswinnerton/pen/XWddGPR, where the arrows can be seen on hover:

<img width="1196" alt="Screen Shot 2020-08-16 at 12 55 36 PM" src="https://user-images.githubusercontent.com/934497/90339592-c8de9780-dfbf-11ea-8f28-d9553ce27678.png">

##

It's worth pointing out that https://github.com/tailwindlabs/tailwindcss/issues/1101 discussed this and the guidance was to create a custom utility to solve this problem. I'm wondering if perhaps `appearance-none` didn't exist then and if it would be considered to add the ability to suppress these arrows in `appearance-none`, or perhaps update the documentation to call out that number-based `input`s are not supported.

**Comment by adamwathan** [comment]
This is just how CSS works, `appearance-none` just sets `appearance: none;` so whatever that does is what the utility does. We could consider deviating from what `appearance: none` actually does and trying to be "smart" about hiding everything we think the user expects to be hidden but I worry those expectations are going to be different for each user :/

**Comment by gcavanunez** [comment]
Yeah is true, it almost feels like its an over all input type sort of deal, since the same renders true with `input` type `search,` where the x is floating where the arrows in an input type number take place. Seem like another sort of utility class would have to used to address this scope of what is desired.

**Comment by binajmen** [comment]
If someone is looking for a solution:
```css
@layer utilities {
  input[type="number"]::-webkit-inner-spin-button,
  input[type="number"]::-webkit-outer-spin-button {
    @apply appearance-none;
  }
}
```

**Reply by EdgarDev17** [reply]
Thank you. This code snippet works fine.

**Reply by MiguelCagnate** [reply]
Confirmed!! thanks a lot👏🏽👏🏽👏🏽👏🏽👏🏽👏🏽

**Reply by md-moijul** [reply]
This doesn't work on Firefox. Does anyone have a solution which is compatible in all major browser? 

update:
```
input[type='number'] {
    -moz-appearance: textfield;
}
```
this does the trick

**Comment by ivnlabs** [comment]
@binajmen thanks, That code makes the work done, also it must go on Tailwinds.css, in Laravel you can find it on the folder called Resources->app.css, copy paste the code there

**Comment by mdjamilkashemporosh** [comment]
@layer base {
  input[type="number"]::-webkit-inner-spin-button,
  input[type="number"]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
}

**Comment by 4anghyeon** [comment]
Tailwind v4

```css
@utility number-input-no-spin {
  @apply [&::-webkit-inner-spin-button]:appearance-none [&::-webkit-outer-spin-button]:appearance-none;
}

```
